### ISSUE TITLE
[Bug] Type errors on using `vtkImageResliceMapper` with `vtkImageSlice`

### ISSUE THREAD

**Issue by jadh4v** [Kitware/vtk-js#2878]
### Bug description

Reported typescript related errors when using `vtkImageResliceMapper` with `vtkImageSlice` (actor).

Bug reported by hmilyma on [VTK discourse](https://discourse.vtk.org/t/error-occurs-when-using-vtkimagereslicemapper/12019/2).

```
Argument of type ‘vtkImageResliceMapper’ is not assignable to parameter of type ‘vtkImageMapper’.
Type ‘vtkImageResliceMapper’ is missing the following properties from type ‘vtkImageMapper’: getSliceAtPosition, getClosestIJKAxis, getBoundsForSlice, getRenderToRectangle, and 15 more.ts(2345)
```

### Steps to reproduce

Recreate the ImageResliceMapper example in typescript to reproduce the error mentioned above.

### Expected Behavior

No type errors

### Environment

- vtk.js version:
- Browsers:
- OS:

**Comment by daker** [comment]
i think `vtkImageSlice` [setMapper](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/ImageSlice/index.d.ts#L131) should accept vtkAbstractImageMapper rather than vtkImageMapper, @floryst what do you think ?

**Comment by floryst** [comment]
Yes, I agree that `setMapper` should take `vtkAbstractImageMapper`. 

We do need to make some adjustments to the typing as well. `vtkImageSlice` depends on the input mapper having `getBoundsForSlice`. Right now that's on `vtkImageMapper`, so we would need to move that to `vtkAbstractImageMapper`.

**Comment by jadh4v** [comment]
I am wondering if `getBoundsForSlice` is actually essential for anything?
`getBoundsForSlice` doesn't appear to be used anywhere except `SliceRepresentationProxy`. There too the call uses default parameters which should be equivalent to calling `getBounds()`. I am assuming that `getBoundsForSlice(model.sliceNumber)` is equivalent to just `getBounds()`.
